src/main/java/org/jenkinsci/plugins/displayurlapi/DisplayURLProvider.java

```diff
@@ -189,6 +193,8 @@ public static DisplayURLProvider getPreferredProvider() {
         if (prefProperty != null && prefProperty.getConfiguredProvider() != null) {
             return prefProperty.getConfiguredProvider();
         }
+        // Note that this logic means that `/display/redirect` links will never be produced when using the environment
+        // variable or system property, meaning that it effetively disables user preferences.
```

Comment:
In other words, the system properties are really a way to disable the effects of this plugin without having to install it.